### ISSUE TITLE
Issue with searching consecutive regions with .search() in same IGV browser since v3.0.0 

### ISSUE THREAD

**Issue by efgstone** [igvteam/igv.js#1864]
I have embedded IGV JS into a web app which pops up within a modal when the user clicks on particular links within the page. Since v3.0.0 I have noticed an issue whereby if a user is searching a larger region of the BAM, say a gene, and then subsequently tries to view a smaller region, either an exon or a SNP for example, the IGV browser instance loads the new location but then instantly reverts back to the previous larger region that was searched.

The user is able to manually search a new region within the IGV browser without issue, but using `igv.browser.search(locus)` within the JavaScript is no longer working as expected.

Interestingly the issue only seems to happen when searching larger regions followed by smaller regions. I can search numerous smaller regions consecutively without issue. The problem was noted when we updated to v3.0.0, and the problem persists with v3.0.1 but it was working fine before with v2.15.11. I have therefore reverted back to this previous version for now.

Below is the snippet of JavaScript which opens the modal and searches for the given position.

```
        function show_igv(locus){
            $('#igvModal').on('shown.bs.modal', function (e) {
                igv.browser.search(locus)
            });
        }
```

**Comment by jrobinso** [comment]
In what way is it failing?  And can you give me a sequence of locations for which it fails.   Note that the search function is asynchronous, if you don't await the results before doing the next search you might get unpredictable results.

Here's my test

```
    const browser = await igv.createBrowser(document.getElementById('igvDiv'), options)

    const loci = ["chr4:48,610,738-144,598,960", "chr4:83,806,419-89,792,781", "chr4:85,608,764-85,953,140"]
    for (let l of loci) {
        await browser.search(l)
        console.log(browser.currentLoci())
    }

```

**Comment by efgstone** [comment]
Hi Jim, thanks very much for looking into this. I was unable to reproduce the issue in a simple codepen, so went back to my webapp and identified an issue with how I had implemented the search functionality. Essentially every time a user was clicking on a new region to show, a new listener event was triggered but was never closed, meaning that all previous look ups were being triggered with every new one. This was an undetected bug until we upgraded to v3.0.0, so not sure why it seemingly worked OK with the older version, but it is good that we have now identified the cause and have fixed the problem. We are now using v3.0.1 without issue. Thanks again for your help.
